test/lit.cfg.py

```diff
@@ -73,6 +73,7 @@
 else:
     config.substitutions.append(('spirv-val', ':'))
 
-if using_spirv_tools:
+env = config.environment
+if using_spirv_tools and ('LD_LIBRARY_PATH' in env):
     new_ld_library_path = os.path.pathsep.join((config.spirv_tools_lib_dir, config.environment['LD_LIBRARY_PATH']))
     config.environment['LD_LIBRARY_PATH'] = new_ld_library_path
```

Comment:
What if `config.spirv_tools_lib_dir` contains a correct path, but `LD_LIBRARY_PATH` is not set? In that case we won't set the proper `LD_LIBRARY_PATH` for SPIR-V Tools and it could lead to errors in some valid environments.

I think that the original code is not perfect and it should be refactored, which will also help you to fix the issue: we should use existing ` llvm_config.with_system_environment` and `llvm_config.with_environment` functions. For example you can take a look how they are used in [intel/llvm repo](https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L63)

Reply:
Thank you for pointing out the example in intel/llvm repo. It is a much nicer approach to check the environment. I changed my commit and please take a look.